lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -1883,6 +1890,31 @@ void OCLToSPIRVBase::visitSubgroupAVCBuiltinCallWithSampler(
       &Attrs);
 }
 
+void OCLToSPIRVBase::visitCallLdexp(CallInst *CI, StringRef MangledName,
+                                    StringRef DemangledName) {
+  auto Args = getArguments(CI);
+  if (Args.size() == 2) {
+    Type *Type0 = Args[0]->getType();
+    Type *Type1 = Args[1]->getType();
+    // For OpenCL built-in math functions 'floatn ldexp(floatn x, int k)' and
+    // 'doublen ldexp(doublen x, int k)', convert scalar arg to vector to keep
+    // consistency with SPIRV spec. Regarding to SPIRV OpenCL Extended
+    // Instruction set, k operand must have the same component count as Result
+    // Type and x operands
+    if (auto *FixedVecType0 = dyn_cast<FixedVectorType>(Type0)) {
+      auto ScalarTypeID = Type0->getScalarType()->getTypeID();
+      if ((ScalarTypeID == llvm::Type::FloatTyID ||
```

Comment:
Should we also handle Half type here? It exists as an extension and can be found here: https://www.khronos.org/registry/OpenCL/specs/2.2/html/OpenCL_Ext.html

Reply:
done, thanks.